pwiz_tools/BiblioSpec/src/ProteinPilotReader.cpp

```diff
@@ -487,20 +494,24 @@ void ProteinPilotReader::addElement(double& mass, string element, int count){
     mass += (count * newMass);
 }
 
+void ProteinPilotReader::initializeMod() {
+    curMod_.name.clear();
+    curMod_.deltaMass = 0;
+}
+
 void ProteinPilotReader::addMod(){
     // first check to see if we already have one for this mod
     map<string, double>::iterator found = modTable_.find(curMod_.name);
     if( found != modTable_.end() && //if it was found and has different mass
         found->second != modTable_[curMod_.name] ){
         throw BlibException(false, "Two entries for a modification named %s,"
-                            "one with delta mass %d and one with %d.",
-                            found->second, modTable_[curMod_.name]);
+                            "one with delta mass %f and one with %f.",
+                            curMod_.name, found->second, modTable_[curMod_.name]);
```

Comment:
I was having trouble reproducing this error with /RTCs; I had to attach a debugger to understand why the error wasn't triggering. Turns out that, ahem, `found->second` is always equal to modTable_[curMod_.name] because we got the found itr by searching modTable_ for that name. :) It should be `found->second != curMod_.deltaMass`. So now I'm wondering why this error would EVER trigger with the current logic. Floating point instability perhaps? The uninitialized values usually have extreme exponents.

Reply:
It's amazing how long you can stare at code and not see seemingly obvious errors... different floating point behavior under VM may explain why we saw the error on TC but not our bare metal machines.